decidim-core/lib/decidim/map/autocomplete.rb

```diff
@@ -9,24 +9,68 @@ class Autocomplete < Map::Frontend
-        # @return [String] The field element's markup.
+        # @param object_name [String, Symbol] The name of the object the field is being generated for.
+        # @param method [String, Symbol] The specific attribute or property of the object.
+        # @param options [Hash] Additional options for customizing the field's behavior and appearance.
+        # @return [String] The HTML markup for the geocoding field.
         def geocoding_field(object_name, method, options = {})
           options[:autocomplete] ||= "off"
-
           append_assets
+
+          if show_my_location_button?
+            geocoding_field_with_location_button(object_name, method, options)
+          else
+            geocoding_field_without_location_button(object_name, method, options)
+          end
+        end
+
+        private
+
+        def geocoding_field_with_location_button(object_name, method, options)
+          template.snippets.add(:decidim_geocoding_scripts, template.append_javascript_pack_tag("decidim_geocoding"))
+          template.snippets.add(:decidim_geocoding_styles, template.append_stylesheet_pack_tag("decidim_geocoding"))
+
+          template.content_tag(:div, class: "geocoding-container") do
+            template.text_field(
+              object_name,
+              method,
+              options.merge("data-decidim-geocoding" => view_options.to_json)
+            ) +
+              template.content_tag(:div, class: "input-group-button user-device-location") do
+                template.content_tag(:button, class: "button button__sm md:button__sm button__text-secondary", type: "button", data: {
+                                       input: "#{object_name}_#{method}",
+                                       latitude: "#{object_name}_latitude",
+                                       longitude: "#{object_name}_longitude",
+                                       error_no_location: I18n.t("errors.no_device_location", scope: "decidim.proposals.forms"),
+                                       error_unsupported: I18n.t("errors.device_not_supported", scope: "decidim.proposals.forms"),
+                                       url: Decidim::Core::Engine.routes.url_helpers.locate_path
+                                     }) do
+                  icon("map-pin-line", role: "img", "aria-hidden": true) + " #{I18n.t("use_my_location", scope: "decidim.proposals.forms")}"
+                end
+              end
+          end
+        end
+
+        def geocoding_field_without_location_button(object_name, method, options)
           template.text_field(
             object_name,
             method,
             options.merge("data-decidim-geocoding" => view_options.to_json)
           )
         end
+
+        def show_my_location_button?
+          return unless template.respond_to?(:current_component)
+
+          Decidim.show_my_location_button.include?(template.current_component.manifest_name.to_sym)
+        end
```

Comment:
```suggestion
        def show_my_location_button?
          return true if Decidim.show_my_location_button == :all
          return unless template.respond_to?(:current_component)

          Decidim.show_my_location_button.include?(template.current_component.manifest_name.to_sym)
        end
```
I suggest having an "all" configuration